src/main/java/com/es/phoneshop/model/product/SampleProduct.java

```diff
@@ -1,29 +1,62 @@
 package com.es.phoneshop.model.product;
 
+import com.es.phoneshop.model.product.price.PriceHistory;
+
 import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.time.Period;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
 import java.util.Currency;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 public final class SampleProduct {
 
     private static final Currency USD = Currency.getInstance("USD");
 
-    public SampleProduct(ProductDao productDao) {
-        createSampleProductsArrayList(productDao);
+    private static List<LocalDate> createLocalDateList() {
+        List<LocalDate> localDates = new ArrayList<>();
+        LocalDate localDate = LocalDate.now();
+        localDate.datesUntil(localDate.minus(3, ChronoUnit.YEARS), Period.ofDays(-365)).forEach(id -> {
+            localDates.add(id);
+        });
+        return localDates;
+    }
+
+    private static List<BigDecimal> createPriceList(BigDecimal productPrice) throws NullPointerException {
+        if (productPrice == null) {
+            throw new NullPointerException("Product price is null");
+        }
+        return Stream.iterate(productPrice, price -> {
+            BigDecimal newPrice
+                    = new BigDecimal(productPrice.intValue() - productPrice.intValue() / 2
```

Comment:
Are you sure that this operation will not eliminate accurate values of BigDecimal?